os/pm/pm_changestate.c

```diff
@@ -290,26 +294,35 @@ int pm_changestate(int domain_indx, enum pm_state_e newstate)
 	if (newstate != PM_RESTORE) {
 		ret = pm_prepall(domain_indx, newstate);
 		if (ret != OK) {
-			/* One or more drivers is not ready for this state change.  Revert to
+			/* One or more drivers is not ready for this state change. Revert to
 			* the preceding state.
 			*/
-
 			newstate = g_pmglobals.domain[domain_indx].state;
-			pm_changeall(domain_indx, newstate);
+
+			/* Revert the preparation to the old state to retain it */
+			pm_prepall(domain_indx, newstate);
+
+			/* Set the recommended state for the domain to the current state, the recommendation will change
+			 * automatically as per PM logic after a pre-specified delay time.
+			 */
+			g_pmglobals.domain[domain_indx].recommended = newstate;
```

Comment:
Hi Amogh,
I believe this is the key point to refresh the state countdown again, I have tested and seems working fine. Thank you for identifying this! I will update these to our working repo.